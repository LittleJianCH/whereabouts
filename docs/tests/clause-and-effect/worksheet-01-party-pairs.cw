Male { name: "bertram" }
Male { name: "percival" }

Female { name: "lucinda" }
Female { name: "camilla" }

Pair { male, female }
------------------------ {
  Male { name: male }
  Female { name: female }
}

// success {
//   Pair { male: "bertram", female: "lucinda" }
// }

// failure {
//   Pair { male: "apollo", female: "daphne" }
// }

query (female) {
  Pair { male: "percival", female }
}

query (female) {
  Pair { male: "camilla", female }
}

query (male) {
  Pair { male, female: "lucinda" }
}

query (x) {
  Pair { male: x, female: x }
}

query (x) {
  Pair { male, female: "fido" }
}

query (male, female) {
  Pair { male, female }
}
