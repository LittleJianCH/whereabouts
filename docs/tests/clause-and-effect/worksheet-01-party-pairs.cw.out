{ "solutions": [{ "female": "lucinda" }, { "female": "camilla" }] }
{ "solutions": [] }
{ "solutions": [{ "male": "bertram" }, { "male": "percival" }] }
{ "solutions": [] }
{ "solutions": [] }
{ "solutions": [{ "male": "bertram", "female": "lucinda" }, { "male": "bertram", "female": "camilla" }, { "male": "percival", "female": "lucinda" }, { "male": "percival", "female": "camilla" }] }
